scripts/cmake/vcpkg_configure_meson.cmake

```
@@ -101,15 +101,15 @@ function(vcpkg_internal_meson_generate_native_file_config _config) #https://meso
     set(LIBPATH_${_config} "'${L_FLAG}${CURRENT_INSTALLED_DIR}${PATH_SUFFIX_${_config}}/lib'")
     
     set(NATIVE_${_config} "[properties]\n") #https://mesonbuild.com/Builtin-options.html
-    string(REGEX REPLACE "( |^)(-|/)" ";\\2" MESON_CFLAGS_${_config} "${VCPKG_DETECTED_CMAKE_C_FLAGS_${_config}}")
+    string(REGEX REPLACE "( +|^)(-|/)" ";\\2" MESON_CFLAGS_${_config} "${VCPKG_DETECTED_CMAKE_C_FLAGS_${_config}}")
```

```suggestion
    string(REGEX REPLACE [=[( +|^)([^"]*("([^\"]|\\|\")*")?)]=] ";\\2" MESON_CFLAGS_${_config} "${VCPKG_DETECTED_CMAKE_C_FLAGS_${_config}}")
```
I think this will match quoted strings with backslash escapes, including forms like `-I"hello world"`, but more testing is required.

---

Will try it out here. Sorry my regex game is pretty weak.

---

Making that change threw this:

```
[DEBUG] CMake Error at scripts/cmake/vcpkg_configure_meson.cmake:104:
[DEBUG]   Parse error.  Expected a newline, got identifier with text "list".
[DEBUG] Call Stack (most recent call first):
[DEBUG]   scripts/ports.cmake:95 (include)
[DEBUG] 
[DEBUG] 
[DEBUG] cmd_execute_and_stream_data() returned 256 after     5939 us
[DEBUG] 
[DEBUG] Compiler information tracking can be disabled by passing --feature-flags=-compilertracking
```

---

Adding in a `+` quantifier seemed to catch the flags with leading `--`

```suggestion
    string(REGEX REPLACE "( +|^)(-+|/)" ";\\2" MESON_CFLAGS_${_config} "${VCPKG_DETECTED_CMAKE_C_FLAGS_${_config}}")
```

---

the error is in get-cmake-vars not here

---

seems to be that the CMake docs are off here a bit. Seems like ^ also matches the direct beginning of the rest of the input after a match.

---

I think this is a strange case where ^ means start of string and start after a matched string. That actually explains a lot.... and CMakes docs seems to be off here which says ^ matches only start of input.
So `string(REGEX REPLACE "( |^)(-)" ";\\1" somevar " ----")` actually results in `;-;-;-;-` instead of the expected `;----`

---

https://gitlab.kitware.com/cmake/cmake/-/issues/16899

---

@ras0219 Your regex can match an empty string due to the combination of  `^` `*` and `?`